keps/sig-network/3766-referencegrant/README.md

```diff
@@ -0,0 +1,492 @@
+- [ ] (R) KEP approvers have approved the KEP status as `implementable`
+- [ ] (R) Design details are appropriately documented
+- [ ] (R) Test plan is in place, giving consideration to SIG Architecture and SIG Testing input (including test refactors)
+  - [ ] e2e Tests for all Beta API Operations (endpoints)
+  - [ ] (R) Ensure GA e2e tests meet requirements for [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md) 
+  - [ ] (R) Minimum Two Week Window for GA e2e tests to prove flake free
+- [ ] (R) Graduation criteria is in place
+  - [ ] (R) [all GA Endpoints](https://github.com/kubernetes/community/pull/1806) must be hit by [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md) 
+- [ ] (R) Production readiness review completed
+- [ ] (R) Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+[ReferenceGrant](https://gateway-api.sigs.k8s.io/api-types/referencegrant/) was
+developed by the [Gateway API subproject](https://gateway-api.sigs.k8s.io/) to
+enable certain object references to cross namespaces. More recently, it has also
+been [used by
+sig-storage](https://kubernetes.io/blog/2023/01/02/cross-namespace-data-sources-alpha/)
+to enable cross-namespace data sources.
+
+This KEP proposes moving ReferenceGrant from its current
+`gateway.networking.k8s.io` API group to a new `grants.authorization.k8s.io` API
+group.
+
+## Motivation
+
+Now that it's clear that ReferenceGrant is useful beyond just Gateway API, it
+would be good to formalize this model in a more neutral home, ideally sig-auth.
+At this point, each project that wants to enable cross-namespace references has
+to choose between introducing a dependency on Gateway API and creating a new
+resource that would largely duplicate ReferenceGrant. Both options would lead to
+confusion for Kubernetes users.
+
+### Goals
+
+* Move ReferenceGrant to sig-auth API Group
+* Clearly define how ReferenceGrant should be used, including both current use
+  cases and guidance for future use cases
+* Implement a library to ensure that ReferenceGrant is implemented consistently
+  by all controllers
+
+### Non-Goals
+
+* Add, remove, or change fields in the ReferenceGrant API
+* Develop an authorizer that will automatically implement ReferenceGrant for
+  all use cases
```

Comment:
I think that would be incompatible with some of the requests we're seeing elsewhere in this KEP, such as "all namespaces" - I think label selectors would be especially problematic for this kind of approach.

Reply:
RBAC does support `*` but not selectors.  This might still be worth discussing though because it could drastically reduce code complexity outside of core.